data/functions.sql

```diff
@@ -1051,3 +1051,31 @@ BEGIN
   RETURN 17;
 END;
 $$ LANGUAGE plpgsql IMMUTABLE;
+
+-- return the capacity of a parking lot, or estimate it from the area and building levels.
+CREATE OR REPLACE FUNCTION tz_estimate_parking_capacity(capacity TEXT, parking TEXT, levels TEXT, way_area REAL)
+RETURNS INTEGER AS $$
+DECLARE
+  levels_int INTEGER;
+  spaces_per_level INTEGER;
+BEGIN
+  -- if the capacity is set, then use that.
+  IF capacity ~ '^[0-9]+$' THEN
+    RETURN capacity::integer;
+  END IF;
+  -- otherwise, try to use the information we have to guess the capacity
+  spaces_per_level := (way_area / 46.0)::integer;
```

Comment:
Just for sanity check:
https://ag.tennessee.edu/cpa/Information%20Sheets/CPA%20222.pdf

1 acre = 4046 sq meters; 

> If a field that is 180 feet by 242 feet (approximately 1 acre) is designed with six rows of parking spaces with each parking space being approximately 10 feet by 18 feet and the traffic lanes are 24 feet wide, approximately 150 spaces can be designed. 

So that's 4046 sq meters / 150 spaces = 27 sq meters per space for a parking garage type situation. 

> Many fields used for parking are not flat, not perfectly square and do not accommodate lined parking spaces. Accounting for these and other imperfections in farm fields that are often used for customer parking, many on-farm marketers often estimate that they can park between 80 and 100 automobiles per acre. 

So that's 4046 sq meters / 89 spaces = 45 sq meters per space for bigger surface parking lot situation.

Reply:
So if it's a parking garage can fit up to 2x as many cars as surface lot (even more if at diagonal, which is rare), all things considered. 

But the point is just to scale it based on something, and this is a reasonable :)

Reply:
Yup, there's a wide range in the existing data. At the moment, the capacity estimate probably has some pretty large error bars. We can fix it so that `parking != surface` implies 27 square meters instead of 45, but we should also include a projection correction. Because `way_area` is in square Mercator meters, parking lots will seem higher capacity at higher latitudes. Follow-up in #1768.